secant/Features/BackupFlow/Views/RecoveryPhraseBackupValidationView.swift

```diff
@@ -0,0 +1,280 @@
+//
+//  RecoveryPhraseBackupView.swift
+//  secant-testnet
+//
+//  Created by Francisco Gindre on 10/29/21.
+//
+
+import SwiftUI
+import ComposableArchitecture
+
+struct RecoveryPhraseBackupValidationView: View {
+    let store: RecoveryPhraseValidationStore
+
+    var body: some View {
+        WithViewStore(self.store) { viewStore in
+            VStack {
+                header(for: viewStore)
+                    .padding(.horizontal)
+                    .padding(.bottom, 10)
+
+                ZStack {
+                    Asset.Colors.BackgroundColors.phraseGridDarkGray.color
+                        .edgesIgnoringSafeArea(.bottom)
+
+                    VStack(spacing: 35) {
+                        let state = viewStore.state
+                        let groups = state.phrase.toGroups()
+
+                        ForEach(Array(zip(groups.indices, groups)), id: \.0) { index, group in
+                            WordChipGrid(
+                                state: state,
+                                groupIndex: index,
+                                wordGroup: group,
+                                misingIndex: index
+                            )
+                            .background(Asset.Colors.BackgroundColors.phraseGridDarkGray.color)
+                            .whenIsDroppable(!state.groupCompleted(index: index), dropDelegate: state.dropDelegate(for: viewStore, group: index))
+                        }
+
+                        Spacer()
+                    }
+                    .padding()
+                    .padding(.top, 0)
+                    .navigationLinkEmpty(
+                        isActive: viewStore.bindingForRoute(.success),
```

Comment:
Uhmmm I'm not sure if I follow. 
This is the custom binding is referring to. 
````Swift
 func bindingForRoute(_ route: RecoveryPhraseValidationState.Route) -> Binding<Bool> {
        self.binding(
            get: { $0.route == route },
            send: { isActive in
                return .updateRoute(isActive ? route : nil)
            }
        )
    }
````

Reply:
Yeah I saw the function, I didn't spend any time trying to see if anything else could be done but it was just a little strange to have two navigation links stacked on one view, when the isactive state and the destination use the same enum.  I thought maybe it would be possible to collapse this down to a single link by using a case path or another mechanism so that you could have a single generic function instead of stacking two.